src/app/users/auth.service.ts

```diff
@@ -126,4 +127,20 @@ export class AuthService {
     return this.db.doc<Question>(`questions/${id}`).valueChanges()
       .pipe(flatMap(q => this.isAdminForCourse(q.course)));
   }
+
+  updateFavoriteCourse(course: number, favorite: boolean) {
+    this.user$.pipe(take(1)).toPromise().then(user => {
+      let newFavorites = user.favoriteCourses;
+      if (favorite) {
+        if (newFavorites.includes(course) === false) {
+          newFavorites.push(course);
+        }
+      } else {
+        newFavorites = newFavorites.filter(c => c !== course);
+      }
+      this.db.doc<UserData>(`users/${user.uid}`).update({
```

Comment:
You ignore a promise here. Please return the promise for the component to be able to show a toast or something when it is done

Reply:
Done.